src/test/java/com/list/todo/services/UserServiceTest.java

```diff
@@ -0,0 +1,242 @@
+
+    @InjectMocks
+    private UserService userService;
+
+    private static final Long currentUserId = 1L;
+
+    @Test
+    public void getUserById_OnExistentUser_ReturnsOptionalOfUser() {
+        //arrange
+        User user = createUser(1);
+        when(userRepository.findById(currentUserId)).thenReturn(Optional.of(user));
+
+        //act
+        Optional<User> returnedUser = userService.getUserById(currentUserId);
+
+        //assert
+        verify(userRepository).findById(currentUserId);
+        Assert.assertEquals(user, returnedUser.get());
+    }
+
+    @Test
+    public void getUserByUsername_OnExistentUser_ReturnsOptionalOfUser() {
+        //arrange
+        User user = createUser(1);
+        when(userRepository.findByUsername(user.getUsername())).thenReturn(Optional.of(user));
+
+        //act
+        Optional<User> returnedUser = userService.getUserByUsername(user.getUsername());
+
+        //assert
+        verify(userRepository).findByUsername(user.getUsername());
+        Assert.assertEquals(user, returnedUser.get());
+    }
+
+    @Test
+    public void getUserInfo_OnExistentUser_ReturnsAnObjectOfUserSummary() {
+        //arrange
+        User user = createUser(1);
+        UserSummary userSummary = new UserSummary(
+                user.getUsername(),
+                user.getName(),
+                user.getEmail(),
+                user.getGravatarHash());
+        UserPrincipal userPrincipal = new UserPrincipal(
+                currentUserId,
+                user.getName(),
+                user.getUsername(),
+                user.getEmail(),
+                user.getPassword(),
+                Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER")));
+        when(userRepository.findById(userPrincipal.getId())).thenReturn(Optional.of(user));
+
+        //act
+        UserSummary returnedUserSummary = userService.getUserInfo(userPrincipal);
+
+        //assert
+        verify(userRepository).findById(userPrincipal.getId());
+        Assert.assertEquals(userSummary.getName(), returnedUserSummary.getName());
+        Assert.assertEquals(userSummary.getEmail(), returnedUserSummary.getEmail());
+        Assert.assertEquals(userSummary.getUsername(), returnedUserSummary.getUsername());
+    }
+
+    @Test
+    public void getUserStats_OnExistentUser_ReturnsAnObjectOfUserStats() {
+        //arrange
+        UserPrincipal userPrincipal = new UserPrincipal(
+                currentUserId,
+                "name",
+                "username",
+                "email",
+                "password",
+                Collections.singletonList(new SimpleGrantedAuthority("ROLE_USER")));
+        List<Share> shares = new ArrayList<>();
+        Share share1 = new Share(userPrincipal.getId(), new TodoList());
+        Share share2 = new Share(userPrincipal.getId(), new TodoList());
+        shares.add(share1);
+        shares.add(share2);
+        List<TodoList> sharedTodoLists = shares.stream()
```

Comment:
TodoList objects in this list contain only NULL values. Make them contain some information

Reply:
It will be fixed after merge